packages/utils/src/waitFor.ts

```diff
@@ -51,3 +51,45 @@ export function waitFor(condition: () => boolean, opts: WaitForOpts = {}): Promi
     };
   });
 }
+
+interface ElapsedTimeTrackerAttributes {
+  msSinceLastError?: number;
+  msSinceLastCall?: number;
+  now: number;
+}
+
+interface ElapsedTimeTrackerOptions {
+  minElapsedTime: number;
+  onError?: (data: ElapsedTimeTrackerAttributes) => void;
+}
+
+type ElapsedTimeTrackerCaller = (data: ElapsedTimeTrackerAttributes) => void;
+export type ElapsedTimeTracker = (fn: ElapsedTimeTrackerCaller) => void;
+
+/**
+ * Create a tracker which keeps track of the last time a function was called
+ *
+ * @param durationMs
+ * @returns
+ */
+export function waitForElapsedTime({minElapsedTime, onError}: ElapsedTimeTrackerOptions): ElapsedTimeTracker {
```

Comment:
```suggestion
export function createCallTracker(): ElapsedTimeTracker {
```
keep it simple, onError doesn't seemed to be being used, add functionality on usage basis

also pass `minElapsedTime` as options on the elapsedTimeTracker, again for simplicity and possibly dynamism (just an opinion but don't feel strongly about it)